src/vm/functions/mod.rs

```diff
@@ -49,95 +51,128 @@ mod sequences;
+    Begin("begin", ClarityVersion::Clarity1),
+    Hash160("hash160", ClarityVersion::Clarity1),
+    Sha256("sha256", ClarityVersion::Clarity1),
+    Sha512("sha512", ClarityVersion::Clarity1),
+    Sha512Trunc256("sha512/256", ClarityVersion::Clarity1),
+    Keccak256("keccak256", ClarityVersion::Clarity1),
+    Secp256k1Recover("secp256k1-recover?", ClarityVersion::Clarity1),
+    Secp256k1Verify("secp256k1-verify", ClarityVersion::Clarity1),
+    Print("print", ClarityVersion::Clarity1),
+    ContractCall("contract-call?", ClarityVersion::Clarity1),
+    AsContract("as-contract", ClarityVersion::Clarity1),
+    ContractOf("contract-of", ClarityVersion::Clarity1),
+    PrincipalOf("principal-of?", ClarityVersion::Clarity1),
+    AtBlock("at-block", ClarityVersion::Clarity1),
+    GetBlockInfo("get-block-info?", ClarityVersion::Clarity1),
+    ConsError("err", ClarityVersion::Clarity1),
+    ConsOkay("ok", ClarityVersion::Clarity1),
+    ConsSome("some", ClarityVersion::Clarity1),
+    DefaultTo("default-to", ClarityVersion::Clarity1),
+    Asserts("asserts!", ClarityVersion::Clarity1),
+    UnwrapRet("unwrap!", ClarityVersion::Clarity1),
+    UnwrapErrRet("unwrap-err!", ClarityVersion::Clarity1),
+    Unwrap("unwrap-panic", ClarityVersion::Clarity1),
+    UnwrapErr("unwrap-err-panic", ClarityVersion::Clarity1),
+    Match("match", ClarityVersion::Clarity1),
+    TryRet("try!", ClarityVersion::Clarity1),
+    IsOkay("is-ok", ClarityVersion::Clarity1),
+    IsNone("is-none", ClarityVersion::Clarity1),
+    IsErr("is-err", ClarityVersion::Clarity1),
+    IsSome("is-some", ClarityVersion::Clarity1),
+    Filter("filter", ClarityVersion::Clarity1),
+    GetTokenBalance("ft-get-balance", ClarityVersion::Clarity1),
+    GetAssetOwner("nft-get-owner?", ClarityVersion::Clarity1),
+    TransferToken("ft-transfer?", ClarityVersion::Clarity1),
+    TransferAsset("nft-transfer?", ClarityVersion::Clarity1),
+    MintAsset("nft-mint?", ClarityVersion::Clarity1),
+    MintToken("ft-mint?", ClarityVersion::Clarity1),
+    GetTokenSupply("ft-get-supply", ClarityVersion::Clarity1),
+    BurnToken("ft-burn?", ClarityVersion::Clarity1),
+    BurnAsset("nft-burn?", ClarityVersion::Clarity1),
+    GetStxBalance("stx-get-balance", ClarityVersion::Clarity1),
+    StxTransfer("stx-transfer?", ClarityVersion::Clarity1),
+    StxBurn("stx-burn?", ClarityVersion::Clarity1),
+    StxGetAccount("stx-account", ClarityVersion::Clarity2),
 });
 
-pub fn lookup_reserved_functions(name: &str) -> Option<CallableType> {
+impl NativeFunctions {
+    pub fn lookup_by_name_exact_version(
+        name: &str,
+        version: &ClarityVersion,
+    ) -> Option<NativeFunctions> {
+        NativeFunctions::lookup_by_name(name).and_then(|native_function| {
+            if &native_function.get_version() == version {
+                Some(native_function)
+            } else {
+                None
+            }
+        })
+    }
+
+    pub fn lookup_by_name_before_version(
```

Comment:
Could rename to `lookup_by_name_at_version`.

Reply:
Yep -- renamed in ab8204133